src/main.rs

```diff
@@ -167,24 +168,35 @@ fn main() {
              .help("Set the API key for the current repository's domain")
              .required(false)
              .takes_value(true))
+        .arg(Arg::with_name("REMOTE_NAME")
+             .long("use-remote")
+             .help("Specify a different remote than 'origin' to be used")
+             .required(false)
+             .takes_value(true))
         .group(ArgGroup::with_name("FLAGS")
                .args(&["NEW_PROJECT_ID", "LIST_MR", "CLEAR_PROJECT_ID", "CLEAR_DOMAIN_KEY"]))
         .arg(Arg::with_name("REQUEST_ID")
              .required(true)
              .conflicts_with_all(&["FLAGS"])
              .index(1))
         .get_matches();
+
+    let remote_name = matches.value_of("REMOTE_NAME").unwrap_or("origin");
+
     if let Some(project_id) = matches.value_of("NEW_PROJECT_ID") {
         set_project_id(project_id);
```

Comment:
Given the `Remote` is derived from the remote URL, and the project ID is a per-remote thing, we should probably migrate to using a per-remote project id.

Currently, the repositories `.git/config` contains a `req` section with a `projectid` field. We should migrate this to using per-remote subsections.

So,

```ini
[req]
    projectid = 1234
```

becomes

```ini
[req "origin"]
    projectid = 1234

[req "other_remote_name"]
    projectid = 5678
```

Where I'm currently on the fence is what we do about existing cases.  Either

1. we fall back to the `req` section if the remote name is `origin` and there is no `req.origin` subsection, or
2. we migrate legacy configs to the new format by testing for `req.projectid` on startup and upgrading it.

I'm leaning towards #2 because it keeps complexity out of the happy-path runtime.  Thoughts?

Reply:
I like the 2. idea as I do agree that it makes the code cleaner. I should update the PR with this soon.